### PR TITLE
refactor: add template source to generated files

### PR DESCRIPTION
**Pull request by macintoshpie**
~~⚠️ WIP ⚠️ 
The target merge branch should be resolved first before we merge this. This is b/c we have to diff a model updated by @helarga, which this PR would make messier.~~

#### Any background context you want to provide?
Currently, the GMT doesn't provide any information about where generated modelica code comes from. This can make debugging/iterating on model changes very difficult when looking at only the generated codde.

#### What does this PR accomplish?
Adds comments to the generated DistrictEnergySystem about what model the section is for, and what template file was used to generate it.

#### How should this be manually tested?
Run one of the tests, verify that the District/DistrictEnergySystem.mo has these comments

#### Screenshots (if appropriate)
Screen Shot 2021-01-22 at 1 28 39 PM
Screen Shot 2021-01-22 at 1 29 26 PM

### CODE REVIEW

**Comment by Helarga** [comment]
see https://github.com/urbanopt/geojson-modelica-translator/pull/286#discussion_r558370245

**Comment by macintoshpie** [comment]
> see #286 (comment)

@Helarga Are you requesting links to files on github rather than paths inside of the repository?

File: geojson_modelica_translator/model_connectors/districts/templates/DistrictEnergySystem.mot [hunk 1]
```diff
@@ -14,19 +14,49 @@ model DistrictEnergySystem
   parameter Modelica.SIunits.TemperatureDifference delHeaWatTemBui=5;
 
   // Models
-  {% for model_definition in models %}
-    {{ model_definition }}
+{% for model in models %}
+  //
+  // Begin Model Instance for {{ model.id }}
```
Comment (nllong) [inline geojson_modelica_translator/model_connectors/districts/templates/DistrictEnergySystem.mot:19]: nice, documention is always good.

File: geojson_modelica_translator/model_connectors/model_base.py [hunk 2]
```diff
@@ -190,6 +190,13 @@ def modelica_path(self, filename):
         return outputname
 
     def render_instance(self, template_params):
+        """Templates the *_Instance file for the model. The templated result will
```
Comment (nllong) [inline geojson_modelica_translator/model_connectors/model_base.py:193]: 🤘

**Review by nllong — APPROVED** [review]

**Comment by Helarga** [comment]
> > see #286 (comment)
> 
> @Helarga Are you requesting links to files on github rather than paths inside of the repository?

is there a possibility to make paths to different files as hyperlinks either on github or inside the rep ? the same idea is implemented at the MBL, not on github but within the Modelica package itself using html eg. https://github.com/lbl-srg/modelica-buildings/blob/master/Buildings/Fluid/Chillers/AbsorptionIndirectSteam.mo#L315 , I believe it will make it easier for the user/developer to check/modify different GMT Modelica components.

**Comment by macintoshpie** [comment]
@Helarga thanks for the link that's helpful!
To link within the repo/package (ie using the `modelica://` scheme, [ref](https://build.openmodelica.org/Documentation/ModelicaReference.Annotations.Documentation.html)), one issue I see is that we couldn't link to our template models because our template files (e.g. the ones that end with .mot) are not included in the package, so that option is probably out.
We could link to a model's template using github though. Would you mind if I created a separate issue and then PR that included this?

**Comment by Helarga** [comment]
> @Helarga thanks for the link that's helpful!
> To link within the repo/package (ie using the `modelica://` scheme, [ref](https://build.openmodelica.org/Documentation/ModelicaReference.Annotations.Documentation.html)), one issue I see is that we couldn't link to our template models because our template files (e.g. the ones that end with .mot) are not included in the package, so that option is probably out.
> We could link to a model's template using github though. Would you mind if I created a separate issue and then PR that included this?

@macintoshpie yes please, thanks.

**Comment by macintoshpie** [comment]
@Helarga are you good with this for now? Can I go ahead and merge?

**Comment by Helarga** [comment]
> @Helarga are you good with this for now? Can I go ahead and merge?

@macintoshpie yes please, thanks for your efforts.

**Pull request merged**